### PR TITLE
Staircase search using c

### PR DESCRIPTION
**Pull request by ShubhamGupta577**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES
### Description

Added the code for Staircase searching using C language
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5736

### CODE REVIEW

**Comment by ShubhamGupta577** [comment]
@bhav09, Please review this PR and add the required labels to it.

File: C/search/Staircase_Search.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+       So, the row can be skipped and move to next row.
+		2. The key is smaller than the current number, this means, that all the elements in the current column are greater than the key.
+       So, the column can be skipped and move to previous column.
+		3. The key is equal to the current number, this means, the key is found.
+	* If the bounds of the 2D Array are reached, this means, the key is not found.
+*/
+/*
+Time Complexity: O(max(n, m))
+Auxiliary Space: O(1)
+*/
```
Comment (ankitaggarwal23) [inline C/search/Staircase_Search.c]: Shift this TC and SC to the end

File: C/search/Staircase_Search.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+#include<stdio.h>     // For input/output
+#include<stdlib.h>    //Header file for malloc function
```
Comment (ankitaggarwal23) [inline C/search/Staircase_Search.c]: Remove redundant comments

File: C/search/Staircase_Search.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+void staircase_search(int** matrix, int n, int m, int key) {
+
+	// int n = sizeof(a) / sizeof(a[0]);
+	// int m = sizeof(a[0]) / sizeof(a[0][0]);
+
```
Comment (ankitaggarwal23) [inline C/search/Staircase_Search.c]: Remove redundant comments

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**